### DISCUSSION TITLE
Two curves on the same graph inside "Model metrics" tab.

### DISCUSSION THREAD

**Discussion by karlogabo** [mlflow/mlflow#13461] [Q&A]
Hello everyone,

I've been working with MLflow and wanted to ask if it's possible to display multiple metrics (e.g., training and validation loss) on a single plot within the "Model metrics" tab of a run.

For example, during my run I´m logging both the training and validation loss. Entering onto a metric on it´s own and the selecting the remaining one allows me to have both curves on one graph, but on the "Model metrics" tab they appear on their own a but it would be helpful to have them appear on the same plot when required by the user.

Is this already implemented and if so could someone indicate me how to achieve this behavior?

 From one MLops engineer to other thank you very much for your help.

Cheers,
Karlo

**Comment by dylan-kelahealth** [comment]
this can be done in the chart tab. select your run, and click add chart. Select your loss curves for the y axis and step for the x axis.

**Reply by karlogabo** [reply]
Hello, thanks for your answer :) 

Yes, I have done it in the chart tab, but I was wondering if its possible to leave it on the "Models Metric" tab by default when I enter one run

**Reply by dylan-kelahealth** [reply]
I know there was a branch at some point to do this, but it doesn't look like it ever made it to a pull request

**Reply by ismael-el-atifi** [reply]
@dylan-kelahealth is there an API to add a chart programmatically ?
Actually if we could add charts at run level it would do the trick.
A chart inside a run would have access only to the metrics values of this run.

**Comment by nanoric** [comment]
I'm seeking for this too: I want to code how plots are arranged, but not to re-arrange plots using keyboard or mouse everytime I refresh my Web UI. Especially when there are lots of plots in my project.

**Comment by clay-jona** [comment]
Upvoting this. I'm surprised this feature isn't there yet.

**Comment by ismael-el-atifi** [comment]
Indeed, this is a must have feature.
We need a programmatic way to draw some curves on the same plot, for a given run.
For example, this is very useful for deep learning to easily compare training and validation losses/metrics across iterations.

**Comment by nranthony** [comment]
I concur !
I'm confused why this was not assume to be needed in the original design.
Please add :)

**Comment by phcerdan** [comment]
Did you find any workaround for this? Coming from ClearML, it has grouping just by "name" of the plot, which is quite user friendly.
It would be great having a programmatic way to group metrics in a single plot.

**Comment by byander** [comment]
Any news on this feature yet?
This is a basic feature, and I'm not sure why it hasn't been implemented.
